stable/bundle_dep.pony

```diff
@@ -50,3 +52,57 @@ class BundleDepGitHub
     if git_tag isnt None then
       Shell("cd " + root_path() + " && git checkout " + (git_tag as String))
     end
+
+class BundleDepLocalGit
+  let bundle: Bundle
+  let info: JsonObject box
+  let package_name: String
+  let local_path: String
+  let git_tag: (String | None)
+  new create(b: Bundle, i: JsonObject box)? =>
+    bundle       = b
+    info         = i
+    package_name = try info.data("package-name") as String
```

Comment:
Yeah, it would be nice not to have to specify this explicitly in the `bundle.json`, as you noted, so let's brainstorm a bit.

So, if we operate under the assumption that we don't want to state it explicitly in the bundle, our problem statement is something like this: we need to build a subdirectory name that is (1) a safe name for a directory with no unsupported characters, (2) unique, and (3) deterministic based on all or some of the attributes of that item in bundle.

On it's face, this problem statement makes it seem like a good place to use some form of hashing, given the unique+deterministic requirements.  A simple hash rendered as a safe filename would suffice to meet those three requirements, I think, and I would accept that for this PR.

If you wanted to go above-and-beyond, we can add a fourth optional requirement that the name be something as meaningful as possible given the information in the bundle.  Maybe this means we do something like take the `local-path`, scrub all non-alphanumeric character sequences into a single dash for each unsupported sequence, then append the hash for more uniqueness.  For example, `../../work/pony/packages/pony-inspect` might be translated to `-work-pony-packages-pony-inspect-b49cfa882bab45e5`.  I'd accept something like this, either in this PR or in a future separate PR.